lib/core/request.c

```diff
@@ -735,3 +736,22 @@ void h2o_resp_add_date_header(h2o_req_t *req)
     h2o_timestamp_t ts = h2o_get_timestamp(req->conn->ctx, &req->pool);
     h2o_add_header(&req->pool, &req->res.headers, H2O_TOKEN_DATE, NULL, ts.str->rfc1123, strlen(ts.str->rfc1123));
 }
+
+void h2o_send_early_hints(h2o_req_t *req)
+{
+    /* 103 must be sent before h2o_start_response is called*/
+    assert(req->_generator == NULL);
+    assert(req->_ostr_top->next == NULL && req->_ostr_top->send_early_hints != NULL);
+
+    req->res.status = 103;
```

Comment:
I would appreciate it if you could clarify whether the purpose of `h2o_send_early_hints`, `on_send_early_hints`, `send_early_hints` are to send an informational header that is neither 100 or 101, or if it specifically targets handling 103.

In case it is the former, the names of the function should not indicate that it is related to "early_hints", and the caller of this function should set the `req->res.status` to 103.

In the case of the latter, we should not set the status code to 103 here or assert that it is is 103 in the protocol implementations.